### ISSUE TITLE
HyperDriveStep dose not accept ScriptRunConfig as a valid run_confic

### ISSUE THREAD

**Issue by masgra** [Azure/MachineLearningNotebooks#1298]
- **Package Name**: azureml.train.hyperdrive.HyperDriveConfig()
- **Package Version**: azureml-train==1.20.0, azureml-core==1.20.0
- **Operating System**: Windows 10
- **Python Version**: 3.8.5

**Describe the bug**
when I create a pipeline step for hyperdrive and specify the `run_config` parameter as a `ScriptRunConfig` object, initialization of `HyperDriveStep` fails because `ScriptRunConfig` has no attribute 'data_references'.

**To Reproduce**
1. create a ScriptRunConfig object with a training script
```
run_config = ScriptRunConfig( source_directory=source_directory, script="train.py", 
environment= Environment.get(ws, name="xgb-hp"),
    compute_target= compute_cluster)
```

2. create a HyperdriveConfig object with this ScriptRunConfig.
```
hyperdrive_config = HyperDriveConfig(  run_config=run_config, 
    hyperparameter_sampling=bayesian_sampling, 
    primary_metric_name='test-mae-mean',
    primary_metric_goal=PrimaryMetricGoal.MINIMIZE, 
    max_total_runs=60,
    max_concurrent_runs=3
    )
```

3. try to create a HyperDriveStep object of a pipeline with this HyperdriveConfig
```
hyperdrive_step = HyperDriveStep(
        name='storeys:hyperdrive',
        hyperdrive_config=hyperdrive_config,
        estimator_entry_script_arguments=[],
        inputs=[train_dataset.as_input('train')],
        outputs=[params_dataset],
    )
```

**Expected behavior**
HyperDriveStep should hot ask ScriptRunConfig for this atribute
**Screenshots**

this is the Error stack:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-57-26fcafea004b> in <module>
      1 # creates the hyperdirve pipeleine step
----> 2 hyperdrive_step = HyperDriveStep(
      3         name='storeys:hyperdrive',
      4         hyperdrive_config=hyperdrive_config,
      5         estimator_entry_script_arguments=[],

~\.conda\envs\AML-prep-xgb\lib\site-packages\azureml\pipeline\steps\hyper_drive_step.py in __init__(self, name, hyperdrive_config, estimator_entry_script_arguments, inputs, outputs, metrics_output, allow_reuse, version)
    188         if run_config.arguments:
    189             raise ValueError('arguments in run_config should not be provided. Please use arguments in the step')
--> 190         if run_config.data_references:
    191             raise ValueError('data_references in run_config should not be provided. Please use inputs in the step')
    192 

AttributeError: 'ScriptRunConfig' object has no attribute 'data_references'`
```
**Additional context**

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 79bd5e63-af29-ac4a-2689-93746107f8e7
* Version Independent ID: 7af884a7-b7e0-953a-1da9-403aaf3a3b68
* Content: [azureml.pipeline.steps.hyper_drive_step.HyperDriveStep class - Azure Machine Learning Python](https://docs.microsoft.com/en-us/python/api/azureml-pipeline-steps/azureml.pipeline.steps.hyper_drive_step.hyperdrivestep?view=azure-ml-py)
* Content Source: [AzureML-Docset/stable/docs-ref-autogen/azureml-pipeline-steps/azureml.pipeline.steps.hyper_drive_step.HyperDriveStep.yml](https://github.com/MicrosoftDocs/MachineLearning-Python-pr/blob/live/AzureML-Docset/stable/docs-ref-autogen/azureml-pipeline-steps/azureml.pipeline.steps.hyper_drive_step.HyperDriveStep.yml)
* Service: **machine-learning**
* Sub-service: **core**
* GitHub Login: @DebFro
* Microsoft Alias: **debfro**

**Comment by lostmygithubaccount** [comment]
can you run `pip list | grep azureml` to show the versions of other azureml packages?

**Comment by masgra** [comment]
> 
> 
> can you run `pip list | grep azureml` to show the versions of other azureml packages?

```
Package                              Version
------------------------------------ -------------------
adal                                 1.2.5
applicationinsights                  0.11.9
astroid                              2.4.2
attrs                                20.3.0
azure-common                         1.1.26
azure-core                           1.10.0
azure-graphrbac                      0.61.1
azure-identity                       1.4.1
azure-mgmt-authorization             0.61.0
azure-mgmt-containerregistry         2.8.0
azure-mgmt-keyvault                  2.2.0
azure-mgmt-resource                  12.0.0
azure-mgmt-storage                   11.2.0
azureml-automl-core                  1.20.0
azureml-core                         1.20.0
azureml-dataprep                     2.7.3
azureml-dataprep-native              27.0.0
azureml-dataprep-rslex               1.5.0
azureml-dataset-runtime              1.20.0
azureml-defaults                     1.20.0
azureml-model-management-sdk         1.0.1b6.post1
azureml-pipeline                     1.20.0
azureml-pipeline-core                1.20.0
azureml-pipeline-steps               1.20.0
azureml-sdk                          1.20.0
azureml-telemetry                    1.20.0
azureml-train                        1.20.0
azureml-train-automl-client          1.20.0
azureml-train-core                   1.20.0
azureml-train-restclients-hyperdrive 1.20.0
backcall                             0.2.0
backports.tempfile                   1.0
backports.weakref                    1.0.post1
brotlipy                             0.7.0
certifi                              2020.12.5
cffi                                 1.14.4
chardet                              4.0.0
click                                7.1.2
click-plugins                        1.1.1
cligj                                0.7.1
cloudpickle                          1.6.0
colorama                             0.4.4
conda                                4.9.2
conda-package-handling               1.7.2
configparser                         3.7.4
contextlib2                          0.6.0.post1
cryptography                         3.3.1
cycler                               0.10.0
decorator                            4.4.2
dill                                 0.3.3
distro                               1.5.0
docker                               4.4.1
dotnetcore2                          2.1.20
Fiona                                1.8.18
Flask                                1.0.3
fusepy                               3.0.1
GDAL                                 3.1.4
geopandas                            0.8.1
gunicorn                             19.9.0
idna                                 2.10
imbalanced-learn                     0.7.0
ipykernel                            5.3.4
ipython                              7.19.0
ipython-genutils                     0.2.0
isodate                              0.6.0
isort                                5.6.4
itsdangerous                         1.1.0
jedi                                 0.18.0
jeepney                              0.6.0
Jinja2                               2.11.2
jmespath                             0.10.0
joblib                               1.0.0
json-logging-py                      0.2
jsonpickle                           1.5.0
jupyter-client                       6.1.7
jupyter-core                         4.7.0
kiwisolver                           1.3.1
lazy-object-proxy                    1.4.3
liac-arff                            2.5.0
MarkupSafe                           1.1.1
matplotlib                           3.3.3
mccabe                               0.6.1
menuinst                             1.4.16
msal                                 1.8.0
msal-extensions                      0.2.2
msrest                               0.6.19
msrestazure                          0.6.4
munch                                2.5.0
ndg-httpsclient                      0.5.1
numpy                                1.19.3
oauthlib                             3.1.0
olefile                              0.46
pandas                               1.2.0
parso                                0.7.0
pathspec                             0.8.1
patsy                                0.5.1
pickleshare                          0.7.5
Pillow                               8.1.0
pip                                  20.3.3
portalocker                          1.7.1
prompt-toolkit                       3.0.8
pyarrow                              1.0.1
pyasn1                               0.4.8
pycosat                              0.6.3
pycparser                            2.20
Pygments                             2.7.4
PyJWT                                1.7.1
pylint                               2.6.0
pyOpenSSL                            19.1.0
pyparsing                            2.4.7
pyproj                               2.6.1.post1
PyQt5                                5.12.3
PyQt5-sip                            4.19.18
PyQtChart                            5.12
PyQtWebEngine                        5.12.1
PySocks                              1.7.1
python-dateutil                      2.8.1
pytz                                 2020.5
pywin32                              227
pyzmq                                20.0.0
requests                             2.25.1
requests-oauthlib                    1.3.0
Rtree                                0.9.7
ruamel-yaml-conda                    0.15.80
ruamel.yaml                          0.16.12
ruamel.yaml.clib                     0.2.2
scikit-learn                         0.24.1
scipy                                1.6.0
seaborn                              0.11.1
SecretStorage                        3.3.0
setuptools                           49.6.0.post20210108
Shapely                              1.7.1
six                                  1.15.0
statsmodels                          0.12.1
threadpoolctl                        2.1.0
toml                                 0.10.1
tornado                              6.1
tqdm                                 4.56.0
traitlets                            5.0.5
urllib3                              1.26.2
wcwidth                              0.2.5
websocket-client                     0.57.0
Werkzeug                             1.0.1
wheel                                0.36.2
win-inet-pton                        1.1.0
wincertstore                         0.2
wrapt                                1.11.2
xgboost                              1.3.2
```

**Comment by temiwale88** [comment]
Hello masgra, 

Thank you for opening this ticket. I need a resolution as well - thus I'm subscribed to this thread. In the interim, nonetheless, I'm unfortunately using the now deprecated 'Estimator' object to push through this HyperDriveStep workflow: 

`est = Estimator(
    source_directory=train_model_folder,
    compute_target=aml_compute,
    entry_script="train.py",
    environment_definition=sklearn_env
)
  
# Create a HyperDriveConfig using the estimator, hyperparameter sampler, and policy.
hyperdrive_config = HyperDriveConfig(estimator=est,
                                        policy=early_termination_policy,
                                        hyperparameter_sampling=ps, 
                                        primary_metric_name="AUC_weighted",
                                        primary_metric_goal=PrimaryMetricGoal.MAXIMIZE, 
                                        max_total_runs=20, # Be declarative or let this be decided by your parameter search space
                                        max_concurrent_runs=4)
hd_step = HyperDriveStep(
    name=hd_step_name,
    hyperdrive_config=hyperdrive_config,
    estimator_entry_script_arguments=[],
    inputs=[output_split_train.parse_parquet_files(), output_split_test.parse_parquet_files()],
    outputs=[metrics_data, saved_model],
    allow_reuse=True)`

As you can see, I'm using the Estimator object and also referencing it in the **HyperDriveConfig** object using the '**estimator**' parameter. Not the best but hope it helps for now until we get some resolution. 

Thanks!

**Comment by lostmygithubaccount** [comment]
Hello, this is a known issue and the team is actively working on it

**Comment by theyorubayesian** [comment]
Great suggestion, Temi Wale! Following the warning about the deprecation of Estimators, I started to make changes to some projects I maintain and I encountered this error. 

I should add that this error disappears if you submit the HyperDrive Config itself. This is useful in cases where you're not implementing a pipeline.

**Comment by kelsey-emnett** [comment]
Has this been resolved yet? I am running into this issue as well on an ongoing project. It seemed that the Estimator was not working anymore either and as far as I know those are the only two choices.

**Comment by masgra** [comment]
>  It seemed that the Estimator was not working anymore either and as far as I know those are the only two choices.

I think it is not published yet. But Estimator still works for me me with azureml-sdk v1.20.0.

**Comment by kelsey-emnett** [comment]
I was having some issues installing packages with the Estimator using the pip_packages and conda_packages arguments. Switching to a conda dependencies file seemed to fix the issue so I will keep using the Estimator.

**Comment by jeydn** [comment]
I am having the same issue, but using the Estimator works.

**Comment by CakmakMuhammet** [comment]
Hi @lostmygithubaccount 
Have you solved this problem? 
Do we have to Estimator with hyperdrivestep in pipeline now or have you solved data_reference issue with ScriptRunConfig?

**Comment by lostmygithubaccount** [comment]
this was fixed in 1.22, are you still experiencing issues?

**Comment by v-strudm-msft** [comment]
Per the above post, this issue was resolved with a fix last month. We'll now close this issue. Please reopen it if it's still a problem.

**Comment by kamalneel81** [comment]
> this was fixed in 1.22, are you still experiencing issues?

Issue still persists, and as mentioned in the comments, work-around with Estimator works.
